<?xml version="1.0"?>
<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="librarian_apriltag_ros_continuous_node" />
  <arg name="camera_frame" default="head_mount_kinect2_rgb_optical_frame" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find librarian_vision_node)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find librarian_vision_node)/config/tags.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="/librarian/apriltag_camera/image_color" />
    <remap from="camera_info" to="/librarian/apriltag_camera/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
</launch>
